### PR TITLE
Inject latest Durable Object migration state into Miniflare

### PR DESCRIPTION
**Pull request by joshthoward**
## What this PR solves / how to test

Fixes #[insert GH or internal issue number(s)].

## Author has addressed the following

- Tests
  - [ ] TODO (before merge)
  - [X] Included
  - [ ] Not necessary because:
- E2E Tests CI Job required? (Use "e2e" label or ask maintainer to run separately)
  - [X] I don't know
  - [ ] Required / Maybe required
  - [ ] Not required because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [ ] TODO (before merge)
  - [x] Included
  - [ ] Not necessary because:
- Public documentation
  - [ ] TODO (before merge)
  - [ ] Cloudflare docs PR(s): 
  - [X] Not necessary because: Covered by SQLite backed DOs announcement and blog post.

### CODE REVIEW

**Comment by joshthoward** [comment]
@penalosa I think this is ready to go. There's one failing test which seems unrelated to my change, but is only reproducible on Windows. Do we have VMs sitting around that we can validate this on?

File: .changeset/lovely-experts-double.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+"miniflare": minor
+"wrangler": minor
+---
+
+Configure SQLite backed Durable Objects in local dev
```
Comment (penalosa) [inline .changeset/lovely-experts-double.md:6]:
```suggestion
feat: Configure SQLite backed Durable Objects in local dev
```

File: packages/miniflare/CONTRIBUTING.md [hunk 2]
```diff
@@ -112,7 +112,8 @@ If your binding is a wrapped binding with its API implemented in
   Miniflare's `workerd.capnp` TypeScript definitions. Copy the new
   `worker.capnp` to
   [`src/runtime/config/workerd.capnp`](./src/runtime/config/workerd.capnp), run
-  `pnpm -F miniflare capnp:workerd`, then update the handwritten types in
+  `capnpc -o ts <prefix>/workerd.capnp` (`capnpc` requires a version of
+  TypeScript which is incompatible with this project), then update the handwritten types in
```
Comment (penalosa) [inline packages/miniflare/CONTRIBUTING.md:116]: We've fixed this script, so this diff shouldn't needed anymore

File: packages/miniflare/package.json [hunk 3]
```diff
@@ -31,7 +31,6 @@
 	],
 	"scripts": {
 		"build": "node scripts/build.mjs && pnpm run types:build",
-		"capnp:workerd": "capnpc -o ts src/runtime/config/workerd.capnp",
```
Comment (penalosa) [inline packages/miniflare/package.json:34]: We've fixed this script, so this diff shouldn't needed anymore

File: packages/miniflare/src/index.ts [hunk 4]
```diff
@@ -325,6 +326,14 @@ function getDurableObjectClassNames(
 				// If we've already seen this class in this service, make sure the
 				// unsafe unique keys and unsafe prevent eviction values match
 				const existingInfo = classNames.get(className);
+				if (existingInfo?.enableSql !== enableSql) {
+					throw new MiniflareCoreError(
+						"ERR_DIFFERENT_STORAGE_BACKEND",
+						`Different storage backeds defined for Durable Object "${className}" in "${serviceName}": ${JSON.stringify(
```
Comment (penalosa) [inline packages/miniflare/src/index.ts:332]:
```suggestion
						`Different storage backends defined for Durable Object "${className}" in "${serviceName}": ${JSON.stringify(
```

File: packages/wrangler/src/dev/miniflare.ts [hunk 5]
```diff
@@ -658,6 +676,55 @@ export function buildMiniflareBindingOptions(config: MiniflareBindingsConfig): {
...
 
+function getClassNamesWhichUseSQLite(
+	migrations: Config["migrations"] | undefined
+) {
+	const classNameToUseSQLite = new Map<string, boolean>();
+	(migrations || []).forEach((migration) => {
+		migration.deleted_classes?.forEach((deleted_class) => {
+			if (!classNameToUseSQLite.delete(deleted_class)) {
+				throw new UserError(
+					`Cannot apply deleted_classes migration to non-existant class ${deleted_class}`
```
Comment (penalosa) [inline packages/wrangler/src/dev/miniflare.ts:687]:
```suggestion
					`Cannot apply deleted_classes migration to non-existent class ${deleted_class}`
```

File: packages/wrangler/src/dev/miniflare.ts [hunk 6]
```diff
@@ -658,6 +676,55 @@ export function buildMiniflareBindingOptions(config: MiniflareBindingsConfig): {
...
+				);
+			}
+		});
+
+		migration.renamed_classes?.forEach(({ from, to }) => {
+			const useSQLite = classNameToUseSQLite.get(from);
+			if (useSQLite === undefined) {
+				throw new UserError(
+					`Cannot apply renamed_classes migration to non-existant class ${from}`
```
Comment (penalosa) [inline packages/wrangler/src/dev/miniflare.ts:696]:
```suggestion
					`Cannot apply renamed_classes migration to non-existent class ${from}`
```

File: packages/wrangler/src/dev/start-server.ts [hunk 7]
```diff
@@ -258,14 +259,15 @@ export async function startDevServer(
 			config: fakeResolvedInput(startDevWorkerOptions),
 			bundle,
 		});
-
+		props.migrations;
```
Comment (penalosa) [inline packages/wrangler/src/dev/start-server.ts:262]:
```suggestion
```

**Review by penalosa — APPROVED** [review]

**Comment by penalosa** [comment]
This looks generally good! A couple small comments, but otherwise good to go.

**Pull request merged**